packages/next/check-error-codes.js

```diff
@@ -0,0 +1,66 @@
+const path = require('path')
+const fs = require('fs/promises')
+
+const errorsDir = path.join(__dirname, '.errors')
+
+// This script checks for new error codes in .errors directory and consolidates them into errors.json.
+// It will fail if new error codes are found, after consolidating them, to ensure error codes are
+// properly reviewed and committed.
```

Comment:
can't we just write `errors.json` directly within the transform, without doing this whole `.errors` thing? it's already reading it after all. this seems a bit convoluted to me

Reply:
Yea, most of the complexity is caused by concurrent compilation not able to read a single `errors.json` without race condition. That's what motivates this whole "two-phase" strategy.

Reply:
hmmm. i feel like ideally we'd just keep all the errors in memory and write them all out at the end, if everything succeeded, which'd avoid this problem. i don't like that we have to jump through all these hoops, feels like we shouldn't have to. but i also don't wanna block the PR on that, we can try to improve this later

Reply:
Unfortunately, each `transform` has a sandboxed WASM instance, thus no way to share global memory that keeps track of errors...